### DISCUSSION TITLE
Compilation of optional C extensions fails calling rc.exe on Win 10 w/ VS 2017

### DISCUSSION THREAD

**Discussion by wholroyd** [jelmer/dulwich#938] [General]
I am trying to work on getting Git accessible inside an AWS Lambda job to tag commits because of some arcane security rules we have. It seems dulwich could solve my issues and am just trying to get it to work first.

I have a clean Windows 10 machine for the most part, with Visual Studio 2017 installed. There is an issue with installing dulwich on this machine because it is trying to call resources which aren't present on the machine. See this log for more details...

[dulwich.txt](https://github.com/dulwich/dulwich/files/2354212/dulwich.txt)

I found a workaround by copying the rc.exe and rcdll.dll files as suggested in [this StackOverflow](https://stackoverflow.com/questions/14372706/visual-studio-cant-build-due-to-rc-exe) thread. There is a command that can be called to set this up for you without copying files if you look at the second highest answer to resolve it as well.

Figured I would let you know the process has a small hurdle on Windows and this optimization step if there was a way to detect and prevent it in the setup.py logic.

**Comment by jelmer** [comment]
This is a problem with your setup; rc.exe doesn't appear to be available. See e.g. https://stackoverflow.com/questions/14372706/visual-studio-cant-build-due-to-rc-exe for somebody else who ran into the same issue (not with Dulwich).
